protocols/kad/src/handler.rs

```diff
@@ -472,9 +470,18 @@ where
             .push(SubstreamState::InWaitingMessage(connec_unique_id, protocol));
     }
 
-    #[inline]
     fn inject_event(&mut self, message: KademliaHandlerIn<TUserData>) {
         match message {
+            KademliaHandlerIn::Reset(request_id) => {
```

Comment:
Can't we just wait for the requests to be answered or to time out instead?

Reply:
This is currently used when the node receives a request to store a record, but for some reason (e.g. a storage error) is unable to (or refuses to) store it. We can just let the request time out on the remote, which is what was done before but there was a TODO left for it (by me). I thought it would be preferable to signal errors as quick as possible to not cause queries unnecessary delay in the normal case, and since the protocol itself has no explicit error responses, resetting the stream seemed the only option. Does that make sense?

Reply:
Oh I see, I misunderstood the purpose.